pkg/lib/dialogs.tsx

```diff
@@ -136,16 +136,29 @@
 
 import React, { useContext, useRef, useState } from "react";
 
-export const DialogsContext = React.createContext();
-export const useDialogs = () => useContext(DialogsContext);
+export interface Dialogs {
+    show(dialog: React.ReactNode): Promise<unknown>;
+    close(args: unknown): void;
+    reject(err: unknown): void;
+    isActive(): boolean;
+}
 
-export const WithDialogs = ({ children }) => {
+export const DialogsContext = React.createContext<Dialogs | null>(null);
+export const useDialogs = () => {
+    const dialogs = useContext(DialogsContext);
+    if (dialogs === null) {
```

Comment:
The default value is the `null` that we pass in to `createContext()` above.  I'm happy to change it to `undefined`, but I don't think it will bring much...

Note: we can trust that we have this correct because Typescript would be complaining at us otherwise: it correctly infers that the return value of `useDialogs()` must surely be a `Dialogs`, since we excluded the `null` possibility (and there is no other possibility).